sdk/eventhub/Azure.Messaging.EventHubs/src/Core/PartitionResolver.cs

```diff
@@ -93,14 +95,31 @@ public virtual string AssignRoundRobin(string[] partitions)
         ///   consideration.
         /// </remarks>
         ///
+        [SkipLocalsInit]
         private static short GenerateHashCode(string partitionKey)
         {
             if (partitionKey == null)
             {
                 return 0;
             }
 
-            ComputeHash(Encoding.UTF8.GetBytes(partitionKey), seed1: 0, seed2: 0, out uint hash1, out uint hash2);
+            const int MaxStackLimit = 256;
+
+            byte[] sharedBuffer = null;
+            var partitionKeySpan = partitionKey.AsSpan();
+            var encoding = Encoding.UTF8;
+
+            var partitionKeyByteLength = encoding.GetMaxByteCount(partitionKey.Length);
+            var hashBuffer = partitionKeyByteLength <= MaxStackLimit ?
+                stackalloc byte[MaxStackLimit] :
+                sharedBuffer = ArrayPool<byte>.Shared.Rent(partitionKeyByteLength);
+
+            var written = encoding.GetBytes(partitionKeySpan, hashBuffer);
+            ComputeHash(hashBuffer.Slice(0, written), seed1: 0, seed2: 0, out uint hash1, out uint hash2);
+            if (sharedBuffer != null)
```

Comment:
Should we move this to a `finally` block?

Reply:
Since this is perf sensitive I would lean towards leaving as is but can understand if you play the safe card 💳😊

Reply:
Talk me off the ledge.  😄    We've got user-provided input that we're assuming we can treat as UTF-8 bytes.   Failure causes a rented buffer to be orphaned.    However, I can't think of a reasonable way that you could create a normal .NET string that would cause an encoding failure.... you?

Reply:
You and me both.   Let's leave things as-is for now.   We can always guard if we have feedback related to errors.